### PR TITLE
add _get_max_nout and alternative nout mode using 'max_nout'

### PR DESCRIPTION
**Pull request by yasirroni**
Add new feature:
New option of nout to accept `nout='max_nout'` and `_get_max_nout()` to count the max_nout.

### CODE REVIEW

**Comment by blink1073** [comment]
Hi @yasirroni, thanks for the contribution!  Can you please add a test in [`test_usage`](https://github.com/blink1073/oct2py/blob/674bd364f66865ed75b0f557ddd5fd1614ad9de3/oct2py/tests/test_usage.py)?

**Comment by yasirroni** [comment]
Done. Any more? Now I wonder, should 'max_nout' be the default, or let 1 be the default to avoid conflict for older version users.

**Comment by blink1073** [comment]
Yes, we need to leave it at 1 for backwards compat.  What we could do though is use `None` as the default and make it a setting on `Oct2Py` like `convert_to_float`.  I'm happy to take this as-is, or you can add that capability if you like.

**Comment by yasirroni** [comment]
> Yes, we need to leave it at 1 for backwards compat. What we could do though is use `None` as the default and make it a setting on `Oct2Py` like `convert_to_float`. I'm happy to take this as-is, or you can add that capability if you like.

I'm not all familiar with working in open source project. Maybe you can do the rest, thanks anyway. 

ps. please add me to the contributors if possible

**Comment by blink1073** [comment]
We can open an issue about making it configurable.  You can add yourself directly to https://github.com/blink1073/oct2py/blob/master/CONTRIBUTORS.rst :smile:

**Comment by blink1073** [comment]
(as part of this PR)

**Comment by yasirroni** [comment]
The readme PR update is failing the test. How can I fix it?

File: README.rst [hunk 1]
```diff
@@ -28,6 +28,33 @@ MAT files.  Usage is as simple as:
 If you want to run legacy m-files, do not have MATLAB®, and do not fully
 trust a code translator, this is your library.
 
+To get .m function, you need to explicitly add the path to .m file using:
+
+.. code-block:: python
+    >>> from oct2py import import octave
```
Comment (blink1073) [inline README.rst:34]: Need a blank line here (RST is picky)

File: README.rst [hunk 2]
```diff
@@ -28,6 +28,33 @@ MAT files.  Usage is as simple as:
...
+    >>> # to add a folder use:
+    >>> octave.addpath('/path/to/directory')
+    >>> # to add folder with all subfolder in it use:
+    >>> octave.addpath(octave.genpath('/path/to/directory'))
+    ...
+
+To get the output of .m file after setting the path, use:
+
+.. code-block:: python
+    >>> x = np.array([[1, 2], [3, 4]], dtype=float)
```
Comment (blink1073) [inline README.rst:44]: Also here

**Comment by yasirroni** [comment]
Sorry, I always use .md, so i didn't get familiar with .rst. I'll try to
fix it tomorrow, since I need to back to sleep. Thanks for the precious
experience.

**Comment by blink1073** [comment]
No problem!

**Comment by yasirroni** [comment]
Still, I can't fix the readme.rst.

I'm confused since my vs code extension to read RST file seems not working.

**Comment by blink1073** [comment]
You can use http://rst.ninjs.org/ to test it out interactively

**Comment by yasirroni** [comment]
There is no problem [here](http://rst.ninjs.org/?theme=nature#T2N0MlB5OiBQeXRob24gdG8gR05VIE9jdGF2ZSBCcmlkZ2UKPT09PT09PT09PT09PT09PT09PT09PT09PT09PT09PT09PT0KCi4uIGltYWdlOjogaHR0cHM6Ly9iYWRnZS5mdXJ5LmlvL3B5L29jdDJweS5wbmcvCiAgICA6dGFyZ2V0OiBodHRwOi8vYmFkZ2UuZnVyeS5pby9weS9vY3QycHkKCi4uIGltYWdlOjogaHR0cHM6Ly9jb2RlY292LmlvL2dpdGh1Yi9ibGluazEwNzMvb2N0MnB5L2NvdmVyYWdlLnN2Zz9icmFuY2g9bWFzdGVyCiAgOnRhcmdldDogaHR0cHM6Ly9jb2RlY292LmlvL2dpdGh1Yi9ibGluazEwNzMvb2N0MnB5P2JyYW5jaD1tYXN0ZXIKCi4uIGltYWdlOjogaHR0cDovL3BlcHkudGVjaC9iYWRnZS9vY3QycHkKICAgOnRhcmdldDogaHR0cDovL3BlcHkudGVjaC9wcm9qZWN0L29jdDJweQogICA6YWx0OiBQeVBpIERvd25sb2FkIHN0YXRzCgpPY3QyUHkgYWxsb3dzIHlvdSB0byBzZWFtbGVzc2x5IGNhbGwgTS1maWxlcyBhbmQgT2N0YXZlIGZ1bmN0aW9ucyBmcm9tIFB5dGhvbi4KSXQgbWFuYWdlcyB0aGUgT2N0YXZlIHNlc3Npb24gZm9yIHlvdSwgc2hhcmluZyBkYXRhIGJlaGluZCB0aGUgc2NlbmVzIHVzaW5nCk1BVCBmaWxlcy4gIFVzYWdlIGlzIGFzIHNpbXBsZSBhczoKCi4uIGNvZGUtYmxvY2s6OiBweXRob24KCiAgICA+Pj4gb2MgPSBvY3QycHkuT2N0MlB5KCkKICAgID4+PiB4ID0gb2MuemVyb3MoMywzKQogICAgPj4+IHByaW50KHgsIHguZHR5cGUpCiAgICBbWyAwLiAgMC4gIDAuXQogICAgIFsgMC4gIDAuICAwLl0KICAgICBbIDAuICAwLiAgMC5dXSBmbG9hdDY0CiAgICAuLi4KClRvIGdldCAubSBmdW5jdGlvbiwgeW91IG5lZWQgdG8gZXhwbGljaXRseSBhZGQgdGhlIHBhdGggdG8gLm0gZmlsZSB1c2luZzoKCi4uIGNvZGUtYmxvY2s6OiBweXRob24KCiAgICA+Pj4gZnJvbSBvY3QycHkgaW1wb3J0IGltcG9ydCBvY3RhdmUKICAgID4+PiAjIHRvIGFkZCBhIGZvbGRlciB1c2U6CiAgICA+Pj4gb2N0YXZlLmFkZHBhdGgoJy9wYXRoL3RvL2RpcmVjdG9yeScpCiAgICA+Pj4gIyB0byBhZGQgZm9sZGVyIHdpdGggYWxsIHN1YmZvbGRlciBpbiBpdCB1c2U6CiAgICA+Pj4gb2N0YXZlLmFkZHBhdGgob2N0YXZlLmdlbnBhdGgoJy9wYXRoL3RvL2RpcmVjdG9yeScpKQogICAgLi4uCgpUbyBnZXQgdGhlIG91dHB1dCBvZiAubSBmaWxlIGFmdGVyIHNldHRpbmcgdGhlIHBhdGgsIHVzZToKCi4uIGNvZGUtYmxvY2s6OiBweXRob24KCiAgICA+Pj4geCA9IG5wLmFycmF5KFtbMSwgMl0sIFszLCA0XV0sIGR0eXBlPWZsb2F0KQogICAgPj4+ICN1c2Ugbm91dD0nbWF4X25vdXQnIHRvIGF1dG9tYXRpY2FsbHkgY2hvb3NlIG1heCBwb3NzaWJsZSBub3V0CiAgICA+Pj4gb3V0LCBvY2xhc3MgPSBvY3RhdmUucm91bmR0cmlwKHgsbm91dD0yKQogICAgPj4+IGltcG9ydCBwcHJpbnQKICAgID4+PiBwcHJpbnQucHByaW50KFt4LCB4LmR0eXBlLCBvdXQsIG9jbGFzcywgb3V0LmR0eXBlXSkKICAgIFthcnJheShbWzEuLCAyLl0sCiAgICAgICAgICAgIFszLiwgNC5dXSksCiAgICAgICAgZHR5cGUoJ2Zsb2F0NjQnKSwKICAgICAgICBhcnJheShbWzEuLCAyLl0sCiAgICAgICAgICAgIFszLiwgNC5dXSksCiAgICAgICAgJ2RvdWJsZScsCiAgICAgICAgZHR5cGUoJzxmOCcpXQogICAgLi4uCgpJZiB5b3Ugd2FudCB0byBydW4gbGVnYWN5IG0tZmlsZXMsIGRvIG5vdCBoYXZlIE1BVExBQsOCwq4sIGFuZCBkbyBub3QgZnVsbHkKdHJ1c3QgYSBjb2RlIHRyYW5zbGF0b3IsIHRoaXMgaXMgeW91ciBsaWJyYXJ5LgoKRmVhdHVyZXMKLS0tLS0tLS0KCi0gU3VwcG9ydHMgYWxsIE9jdGF2ZSBkYXRhdHlwZXMgYW5kIG1vc3QgUHl0aG9uIGRhdGF0eXBlcyBhbmQgTnVtcHkgZHR5cGVzLgotIFByb3ZpZGVzIE9jdGF2ZU1hZ2ljXyBmb3IgSVB5dGhvbiwgaW5jbHVkaW5nIGlubGluZSBwbG90dGluZyBpbiBub3RlYm9va3MuCi0gU3VwcG9ydHMgY2VsbCBhcnJheXMgYW5kIHN0cnVjdHMvc3RydWN0IGFycmF5cyB3aXRoIGFyYml0cmFyeSBuZXN0aW5nLgotIFN1cHBvcnRzIHNwYXJzZSBtYXRyaWNlcy4KLSBCdWlsZHMgbWV0aG9kcyBvbiB0aGUgZmx5IGxpbmtlZCB0byBPY3RhdmUgY29tbWFuZHMgKGUuZy4gYHplcm9zYCBhYm92ZSkuCi0gVGhyZWFkLXNhZmV0eTogZWFjaCBPY3QyUHkgb2JqZWN0IHVzZXMgYW4gaW5kZXBlbmRlbnQgT2N0YXZlIHNlc3Npb24uCi0gQ2FuIGJlIHVzZWQgYXMgYSBjb250ZXh0IG1hbmFnZXIuCi0gU3VwcG9ydHMgVW5pY29kZSBjaGFyYWN0ZXJzLgotIFN1cHBvcnRzIGxvZ2dpbmcgb2Ygc2Vzc2lvbiBjb21tYW5kcy4KLSBPcHRpb25hbCB0aW1lb3V0IGNvbW1hbmQgcGFyYW1ldGVyIHRvIHByZXZlbnQgcnVuYXdheSBPY3RhdmUgc2Vzc2lvbnMuCgoKLi4gX09jdGF2ZU1hZ2ljOiBodHRwczovL25idmlld2VyLmp1cHl0ZXIub3JnL2dpdGh1Yi9ibGluazEwNzMvb2N0MnB5L2Jsb2IvbWFzdGVyL2V4YW1wbGUvb2N0YXZlbWFnaWNfZXh0ZW5zaW9uLmlweW5iP2NyZWF0ZT0xCgoKSW5zdGFsbGF0aW9uCi0tLS0tLS0tLS0tLQpZb3UgbXVzdCBoYXZlIEdOVSBPY3RhdmUgaW5zdGFsbGVkIGFuZCBpbiB5b3VyIGBgUEFUSGBgIGVudmlyb25tZW50IHZhcmlhYmxlLgpBbHRlcm5hdGl2ZWx5LCB5b3UgY2FuIHNldCBhbiBgYE9DVEFWRV9FWEVDVVRBQkxFYGAgb3IgYGBPQ1RBVkVgYCBlbnZpcm9ubWVudAp2YXJpYWJsZSB0aGF0IHBvaW50cyB0byBgYG9jdGF2ZS1jbGlgYCBleGVjdXRhYmxlIGl0c2VsZi4KCllvdSBtdXN0IGhhdmUgdGhlIE51bXB5IGFuZCBTY2lweSBsaWJyYXJpZXMgZm9yIFB5dGhvbiBpbnN0YWxsZWQuClNlZSB0aGUgaW5zdGFsbGF0aW9uIGluc3RydWN0aW9uc18gZm9yIG1vcmUgZGV0YWlscy4KCk9uY2UgdGhlIGRlcGVuZGVuY2llcyBoYXZlIGJlZW4gaW5zdGFsbGVkLCBydW46CgouLiBjb2RlLWJsb2NrOjogYmFzaAoKICAgICQgcGlwIGluc3RhbGwgb2N0MnB5CgpJZiB1c2luZyBjb25kYSwgaXQgaXMgYXZhaWxhYmxlIG9uIGNvbmRhLWZvcmdlOgoKLi4gY29kZS1ibG9jazo6IGJhc2gKCiAgICQgY29uZGEgaW5zdGFsbCAtYyBjb25kYS1mb3JnZSBvY3QycHkKCi4uIF9pbnN0cnVjdGlvbnM6IGh0dHA6Ly9ibGluazEwNzMuZ2l0aHViLmlvL29jdDJweS9zb3VyY2UvaW5zdGFsbGF0aW9uLmh0bWwKCgpEb2N1bWVudGF0aW9uCi0tLS0tLS0tLS0tLS0KCkRvY3VtZW50YXRpb24gaXMgYXZhaWxhYmxlIG9ubGluZV8uCgpGb3IgdmVyc2lvbiBpbmZvcm1hdGlvbiwgc2VlIHRoZSBSZXZpc2lvbiBIaXN0b3J5Xy4KCi4uIF9vbmxpbmU6IGh0dHBzOi8vb2N0MnB5LnJlYWR0aGVkb2NzLmlvL2VuL2xhdGVzdC8KCi4uIF9IaXN0b3J5OiBodHRwczovL2dpdGh1Yi5jb20vYmxpbmsxMDczL29jdDJweS9ibG9iL21hc3Rlci9ISVNUT1JZLnJzdAo=)

I still didn't see why [Codecov Test fail with No CI provider detected?](https://github.com/blink1073/oct2py/pull/170/checks?check_run_id=943360102)

Sorry, still newbie here.

**Comment by blink1073** [comment]
Ah, right, the codecov failure is not due to this PR.  I'll have to look at it later.

**Comment by yasirroni** [comment]
Okay, thaanks. Wish you all the best. Maybe I will implement my changes on my local repo for now.

**Comment by blink1073** [comment]
Confirmed in #171, I will try and fix the problem there.

**Comment by blink1073** [comment]
I fixed the codecov issue, and re-started the jobs here.

**Comment by yasirroni** [comment]
Still not working. o.O

**Comment by yasirroni** [comment]
Should I make new branch from master and make new PR?

**Comment by blink1073** [comment]
Let's try a reset

**Comment by yasirroni** [comment]
I compile non rst file first, then I will commit those with rst.

**Comment by yasirroni** [comment]
The rst file error still persist. o.O

**Comment by blink1073** [comment]
Released as version 5.2.0.

**Pull request merged**